c7n/filters/costhub.py

```diff
@@ -69,10 +69,11 @@ class CostHubRecommendation(Filter):
     }
 
     permissions = ('cost-optimization-hub:ListRecommendations',)
-    annotation = "c7n:cost_optimize"
+    annotation_key = "c7n:cost_optimize"
 
     def process(self, resources, event=None):
-        client = local_session(self.manager.session_factory).client('cost-optimization-hub')
+        client = local_session(self.manager.session_factory).client(
+            'cost-optimization-hub', region_name='us-east-1')
```

Comment:
the endpoint for hub is only in us-east-1, that endpoint does hold data for other regions